flocker_bb/builders/flocker.py

```diff
@@ -433,10 +435,115 @@ def makeOmnibusFactory(distribution):
+    factory = getFlockerFactory(python="python2.7")
+    factory.addStep(SetPropertyFromCommand(
+        command=["python", "setup.py", "--version"],
+        name='check-version',
+        description=['checking', 'version'],
+        descriptionDone=['checking', 'version'],
+        property='version'
+    ))
+    factory.addSteps(installDependencies())
+
+    # Create suitable names for files hosted on Buildbot master.
+
+    sdist_file = 'Flocker-sdist.tar.gz'
+    sdist_path = Interpolate(
+        '%(kw:path)s/%(kw:file)s',
+        path=resultPath('homebrew'),
+        file=sdist_file)
+    sdist_url = Interpolate(
+        "%(kw:base_url)s%(kw:url)s/%(kw:file)s",
+        base_url=buildbotURL,
+        url=resultURL('homebrew'),
+        file=sdist_file
+    )
+    recipe_file = 'Flockerdev.rb'
+    recipe_path = Interpolate(
+        '%(kw:path)s/%(kw:file)s',
+        path=resultPath('homebrew'),
+        file=recipe_file)
+    recipe_url = Interpolate(
+        "%(kw:base_url)s%(kw:url)s/%(kw:file)s",
+        base_url=buildbotURL,
+        url=resultURL('homebrew'),
+        file=recipe_file
+    )
+
+    # Build source distribution
+    factory.addStep(ShellCommand(
+        name='build-sdist',
+        description=["building", "sdist"],
+        descriptionDone=["build", "sdist"],
+        command=[
+            virtualenvBinary('python'),
+            "setup.py", "sdist",
+            ],
+        haltOnFailure=True))
+
+    # Upload source distribution to master
+    factory.addStep(FileUpload(
+        name='upload-sdist',
+        slavesrc=Interpolate('dist/Flocker-%(prop:version)s.tar.gz'),
+        masterdest=sdist_path
+    ))
+
+    # Build Homebrew recipe from source distribution URL
+    factory.addStep(ShellCommand(
+        name='make-homebrew-recipe',
+        description=["building", "recipe"],
+        descriptionDone=["build", "recipe"],
+        command=[
+            virtualenvBinary('python'), "-m", "admin.homebrew",
+            "--flocker-version", "dev",
+            "--sdist", sdist_url,
+            "--output-file", "Flockerdev.rb"],
+        haltOnFailure=True))
+
+    # Upload new .rb file to BuildBot master
+    factory.addStep(FileUpload(
+        name='upload-homebrew-recipe',
+        slavesrc="Flockerdev.rb",
+        masterdest=recipe_path
+    ))
+
+    # Trigger the homebrew-test build
+    factory.addStep(Trigger(
+        name='trigger-homebrew-test',
+        schedulerNames=['trigger/homebrew-created'],
+        set_properties={
+            'master_recipe': recipe_url
+            },
+        waitForFinish=False,
+        ))
+
+    return factory
+
+
+def makeHomebrewRecipeTestFactory():
+    factory = getFlockerFactory(python="python2.7")
+    factory.addSteps(installDependencies())
+
+    # Run testbrew script
+    recipe_url = Property('master_recipe')
+    factory.addStep(ShellCommand(
+        name='run-homebrew-test',
+        description=["running", "recipe"],
```

Comment:
This is not really running the recipe. The script name is more accurate, as it is running the Homebrew test method.